### PR TITLE
instructions or gpt_assistant description issue resolve

### PR DESCRIPTION
**Pull request by giriprasad51**
Hiii @sonichi @qingyun-wu  I made some changes in code please go through it

### CODE REVIEW

**Comment by giriprasad51** [comment]
suggestion me @sonichi @qingyun-wu 
There is case where candidate_assistants >0 and still instructions = None in this senioria what was the excepted instructions for GPTAssistantAgent
---------------------------------------------------------------
# Create two agents with the required configurations
agent101 = GPTAssistantAgent(
    name="agent1",
    llm_config={
        "config_list": config_list,
    },
    
)
print(agent101.assistant_id,agent101.description)

--------------------------------------------------------------------------------------------------------------------
print("-------------check-point-candidate_assistants-------------------")
print(candidate_assistants)
-------------check-point-candidate_assistants-------------------
[Assistant(id='asst_Yc6bPQLdmNapw9MSynnUV9aE', created_at=1720600525, description=None, instructions="\n    As 'movie director' your primary role is to giving suggestions to all departments head and actors\n    ", metadata={}, model='gpt-4o', name='agent1', object='assistant', tools=[], response_format='auto', temperature=1.0, tool_resources=ToolResources(code_interpreter=None, file_search=None), top_p=1.0), Assistant(id='asst_Z0lHerXqlg82zSBL9WjZn6BM', created_at=1720595479, description=None, instructions="\n    As 'weather reporter' your primary role is to information about weather\n", metadata={}, model='gpt-4o', name='agent1', object='assistant', tools=[], response_format='auto', temperature=1.0, tool_resources=ToolResources(code_interpreter=None, file_search=None), top_p=1.0)]


asst_Yc6bPQLdmNapw9MSynnUV9aE None

**Comment by giriprasad51** [comment]
Hiii @sonichi @qingyun-wu need some clarification 
if instructions == None and GPTAssistantAgent has candidate_assistants at that case what are excepted instructions 
(AssistantAgent.DEFAULT_SYSTEM_MESSAGE or candidate_assistants[0].instructions)

File: autogen/agentchat/contrib/gpt_assistant_agent.py [hunk 1]
```diff
@@ -93,6 +90,7 @@ def __init__(
                     instructions,
                     openai_assistant_cfg.get("tools", []),
                 )
+                instructions = candidate_assistants[0].instructions
```
Comment (sonichi) [inline autogen/agentchat/contrib/gpt_assistant_agent.py:93]: Why is this line needed? Line 174 updates the system message.
I do think that line 173 and 174 can be merged by setting the right system message in line 173 directly, unless
@IANTHEREAL suggests otherwise.

Comment (IANTHEREAL) [reply autogen/agentchat/contrib/gpt_assistant_agent.py:93]: Sorry for the late reply. At lines 71 and 73, we will remove the model from `openai_client_cfg` caused by openai assistant api design. Please confirm if this will cause an error with `super().__init__.` I remember that `ConversableAgent._validate_llm_config` checks the configuration of the model. If it doesn’t cause an error now, please follow @sonichi suggestion.

**Comment by jackgerrits** [comment]
Looks like this is dormant, if you want to pick this up again please feel free to reopen. Thanks!
